doc/developer/workflow.rst

```diff
@@ -539,6 +541,28 @@ your new claim at the end of the list.
      * ...
      */
 
+Defensive coding requirements
+-----------------------------
+
+In general, code submitted into FRR will be rejected if it uses unsafe
+programming practices.  While there is no enforced overall ruleset, the
+following requirements have achieved consensus:
+
+- ``strcpy`` and ``sprintf`` are inacceptable without exception.  Use
+  ``strlcpy`` and ``snprintf`` instead.  (Rationale:  even if you know the
+  operation cannot overflow the buffer, a future code change may inadvertedly
+  introduce an overflow.)
```

Comment:
Ban `strcat` in favor of `strlcat` too, I killed that one last year

Reply:
updated.